### PR TITLE
definitions: Implement GSP-706 Support HTTP Signer

### PR DESCRIPTION
**Pull request by JinnyYi**
ref: https://github.com/beyondstorage/go-storage/issues/707

### CODE REVIEW

File: types/operation.generated.go [hunk 1]
```diff
@@ -274,6 +316,47 @@ func (s UnimplementedFetcher) FetchWithContext(ctx context.Context, path string,
...
 }
 
+// Operation names in HTTPSigner.
+var (
+	// OpHTTPSignerQuerySignHTTP is the operation name for QuerySignHTTP in HTTPSigner.
+	OpHTTPSignerQuerySignHTTP = "QuerySignHTTP"
+)
+
+// HTTPSigner is the interface for Signer.
+type HTTPSigner interface {
```
Comment (JinnyYi) [inline types/operation.generated.go:326]: `http` is considered as a common initialisms and is converted to `HTTP` in interface name and operation name. Should I update GSP for consistency?

Comment (Xuanwo) [reply types/operation.generated.go:326]: LGTM

File: types/operation.generated.go [hunk 2]
```diff
@@ -376,6 +465,18 @@ func (s UnimplementedMover) MoveWithContext(ctx context.Context, src string, dst
 	return
 }
 
+// Operation names in Multiparter.
+var (
+	// OpMultiparterCompleteMultipart is the operation name for CompleteMultipart in Multiparter.
+	OpMultiparterCompleteMultipart = "CompleteMultipart"
```
Comment (Xuanwo) [inline types/operation.generated.go:471]: How about using a snake case like `complete_multipart` instead?

File: cmd/definitions/tmpl/operation.tmpl [hunk 3]
```diff
@@ -3,9 +3,23 @@ package types
...
 {{- range $_, $i := .Interfaces }}
+// Operation names in {{ $i.DisplayName }}.
+const (
+{{- range $_, $op := $i.Ops }}
+    // Op{{ $i.DisplayName }}{{ $op.Name | toPascal }} is the operation name for {{ $op.Name }} in {{ $i.Name }}.
+    Op{{ $i.DisplayName }}{{ $op.Name | toPascal }} = "{{ $op.Name }}"
+{{- end }}
+)
+
+{{- range $_, $op := $i.Ops }}
```
Comment (Xuanwo) [inline cmd/definitions/tmpl/operation.tmpl:19]: I prefer

```go
var (
      _ = xxxxx
      _ = xxxxx
)
```

**Review by Xuanwo — APPROVED** [review]

**Comment by Xuanwo** [comment]
Great work!

**Pull request merged**